src/main/java/DeadlineTask.java

```diff
@@ -0,0 +1,84 @@
+public class DeadlineTask extends Task {
+  /**
+   * Deadline
+   */
+  protected String deadline;
+
+  /**
+   * Constructor for Deadline task.
+   *
+   * @param description Description of task
+   * @param deadline    Deadline for task
+   */
+  public DeadlineTask(String description, String deadline) {
+    super(description);
+    this.deadline = deadline;
+  }
+
+  /**
+   * Constructor for deadline task, specifying completion status.
+   *
+   * @param description Description of task
+   * @param isDone      Boolean representing task completion status
+   * @param deadline    Deadline for task
+   */
+  public DeadlineTask(String description, String deadline, boolean isDone) {
+    super(description, isDone);
+    this.deadline = deadline;
+  }
+
+
+  /**
+   * Returns Deadline task formatted with type and status icon
+   *
+   * @return Task formatted as a string.
+   */
+  @Override
+  public String toString() {
+    return String.format("<D>%s (BY: %s)", super.toString(), this.deadline);
+  }
+
+  /**
+   * Parses save file data into a deadline task instance.
+   *
+   * @param data line from save file
+   * @return Task instance
+   * @throws InvalidTaskDataException if data is not in the expected format
+   */
+  public static DeadlineTask fromData(String data) throws InvalidTaskDataException {
+     /*
```

Comment:
I like how you have descriptive documentation to show the expected format. However, I was wondering if this should be part of the JavaDocs?